WordPress/gutenberg
# RN: Add Icon Button Block Editor component.

## Description
This PR adds the menu-item component into its own component now called "IconButton". 
The main reason for splitting up the component is to be able to reuse it in 3rd party blocks. 

## How has this been tested?
Since this a mobile only component for now it has been tested in the following way. 

It has been tested using the mobile demo app on both Android and iOS. 
1. After clicking the (+) the insert should appear as before. 
2. Select the column block. Notice that the layout column selector looks the same as before. 
3. Select one of the layouts. Notice that it works as before.

## Screenshots 
Andorid
![Screen Shot 2020-09-09 at 4 46 50 PM](https://user-images.githubusercontent.com/115071/92665709-121ec000-f2bc-11ea-9ffb-c20fee1c4a51.png)
![Screen Shot 2020-09-09 at 4 46 32 PM](https://user-images.githubusercontent.com/115071/92665728-1ea31880-f2bc-11ea-945c-e5332caba605.png)

iOS
![Screen Shot 2020-09-09 at 4 45 46 PM](https://user-images.githubusercontent.com/115071/92665731-1fd44580-f2bc-11ea-874b-a1ada3087f9c.png)
![Screen Shot 2020-09-09 at 4 45 01 PM](https://user-images.githubusercontent.com/115071/92665735-206cdc00-f2bc-11ea-8e5c-6f5b1afbc8dc.png)

## Types of changes
This is mostly a refactor so that we can use the same use the new component in 3rd party plugins.

## Checklist:
